packages/govern-contract-utils/contracts/minimal-proxies/ERC1167ProxyFactory.sol

```diff
@@ -41,14 +41,15 @@ library ERC1167ProxyFactory {
         require(ok, _getRevertMsg(ret));
     }
 
-    function generateCode(address _implementation) internal pure returns (bytes memory code) {
-        code = new bytes(55);
-        
-        assembly {
-            mstore(add(code, 0x20), 0x3d602d80600a3d3981f3363d3d373d3d3d363d73000000000000000000000000)
-            mstore(add(code, 0x34), shl(0x60, _implementation))
-            mstore(add(code, 0x48), 0x5af43d82803e903d91602b57fd5bf30000000000000000000000000000000000)
-        }
+    function generateCreateData(address _implementation) internal pure returns (bytes memory) {
+        return abi.encodePacked(
+            //---- constructor -----
+            bytes10(0x3d602d80600a3d3981f3),
+            //---- proxy code -----
+            bytes10(0x363d3d373d3d3d363d73),
+            _implementation,
+            bytes15(0x5af43d82803e903d91602b57fd5bf3)
```

Comment:
Does the new code return the error data on revert?

Reply:
This is the same code in EIP-1167; from the opcode table on the EIP page, I think it should be.

Reply:
Should we make the tests for Govern.sol and GovernQueue.sol run both against the contract directly and the contract behind a proxy as we used to do for some contracts aragonOS and/or aragon-apps?

Reply:
We currently just have unit tests and could extend our contract test suite to include integration tests for the whole govern contract framework. This would automatically include the mentioned additional tests from @izqui. A test suite like this would give us the highest possible proof that anything is working as expected.